### PR TITLE
Polymorphism checks

### PR DESCRIPTION
**Pull request by UlfNorell**
- [PT-168026330](https://www.pivotaltracker.com/story/show/168026330)
- [PT-166785003](https://www.pivotaltracker.com/story/show/166785003)
- [PT-166786491](https://www.pivotaltracker.com/story/show/166786491)
- [PT-168216899](https://www.pivotaltracker.com/story/show/168216899)

Enforces
- No higher-order entrypoints
- No polymorphic entrypoints (AEVM only)
- No polymorphic or higher-order oracles or `AENS.resolve` calls

### CODE REVIEW

File: rebar.lock [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {"1.1.0",
 [{<<"aebytecode">>,
   {git,"https://github.com/aeternity/aebytecode.git",
-       {ref,"10cc1278831ad7e90138533466ceef4bcafd74a9"}},
+       {ref,"e7f2be7ce878b1e22bad287f3342f32579e98599"}},
```
Comment (hanssv) [inline rebar.lock]: I see... To synchronize with `aeternity` repo.

**Comment by radrow** [comment]
Why are polymorphic entrypoints banned?

**Comment by UlfNorell** [comment]
They're only banned in the AEVM where they don't work (unless you instantiate all type variables to `word`).

**Comment by UlfNorell** [comment]
Making sure error messages are nice and uniform is not part of this PR. Currently errors from the FATE backend are not being caught and returned gracefully. @hanssv is working on this.

**Comment by UlfNorell** [comment]
Now also fixes #112.

**Comment by radrow** [comment]
I moved it to PT https://www.pivotaltracker.com/story/show/168216899, may be mentioned instead of #112

**Pull request merged**